tiatoolbox/utils/misc.py

```diff
@@ -1257,4 +1268,58 @@ def patch_pred_store(
     store = SQLiteStore()
     keys = store.append_many(annotations, [str(i) for i in range(len(annotations))])
 
+    # if a save director is provided, then dump store into a file
+    if save_dir and output_file:
+        output_file += ".db"
+        path_to_output_file = save_dir / output_file
+        save_dir.mkdir(parents=True, exist_ok=True)
+        store.dump(path_to_output_file)
+        return path_to_output_file
+
     return store
+
+
+def patch_pred_store_zarr(
+    raw_predictions: dict,
+    save_dir: Path,
```

Comment:
same as above, please replace this with `save_path`, we can pass `save_dir / output_name` from engines.